tests_e2e/tests/scripts/agent_firewall-verify_all_firewall_rules.py

```diff
@@ -0,0 +1,402 @@
+from typing import List
+
+if sys.version_info[0] == 3:
+    import http.client as httpclient
+elif sys.version_info[0] == 2:
+    import httplib as httpclient  # pylint: disable=E0401
+
+NON_ROOT_USER = 'unknown_user'
+ROOT_USER = 'root'
+WIRESERVER_ENDPOINT_FILE = '/var/lib/waagent/WireServerEndpoint'
+WIRESERVER_URL = '168.63.129.16'
+VERSIONS_PATH = '/?comp=versions'
+
+
+def get_wireserver_ip():
+    try:
+        with open(WIRESERVER_ENDPOINT_FILE, 'r') as f:
+            wireserver_ip = f.read()
+    except Exception:
+        wireserver_ip = WIRESERVER_URL
+    return wireserver_ip
+
+
+def switch_user(user):
+    # This function switches the function to a given user
+    try:
+        uid = pwd.getpwnam(user)[2]
+        logging.info("uid:%s and user name:%s", uid, user)
+        os.seteuid(uid)
+    except Exception as e:
+        raise Exception("Error -- failed to switch user to {0} : Failed with exception {1}".format(user, e))
+
+
+def get_root_accept_rule(command: str):
+    return ['sudo', 'iptables', '-t', 'security', command, 'OUTPUT', '-d', get_wireserver_ip(), '-p', 'tcp', '-m',
+            'owner',
+            '--uid-owner',
+            '0', '-j', 'ACCEPT', '-w']
+
+
+def get_non_root_accept_rule(command: str):
+    return ['sudo', 'iptables', '-t', 'security', command, 'OUTPUT', '-d', get_wireserver_ip(), '-p', 'tcp',
+            '--destination-port', '53', '-j',
+            'ACCEPT', '-w']
+
+
+def get_non_root_drop_rule(command: str):
+    return ['sudo', 'iptables', '-t', 'security', command, 'OUTPUT', '-d', get_wireserver_ip(), '-p', 'tcp', '-m',
+            'conntrack', '--ctstate',
+            'INVALID,NEW', '-j', 'DROP', '-w']
+
+
+def execute_cmd(cmd: List[str]):
+    proc = subprocess.Popen(cmd,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            shell=False)
+    stdout, stderr = proc.communicate()
+    return proc.returncode, stdout, stderr
+
+
+def check_if_iptable_rule_is_available(full_command: List[str]):
+    # This function is used to check if given rule is present in iptable rule set
+    # "-C" return exit code 0 if the rule is available.
+    exit_code, _, _ = execute_cmd(full_command)
+    if exit_code == 0:
+        return True
+    return False
+
+
+def print_current_iptable_rules():
+    # This function prints the current iptable rules
+    try:
+        cmd = ['sudo', 'iptables', '-L', 'OUTPUT', '-t', 'security', '-nxv']
+        exit_code, stdout, stderr = execute_cmd(cmd)
+        if exit_code != 0:
+            logging.warning("Warning -- Failed to fetch the ip table rules with error code: %s and error: %s", exit_code, stderr)
+        else:
+            for line in stdout.splitlines():
+                logging.info(str(line))
+    except Exception as error:
+        raise Exception("Error -- Failed to fetch the ip table rule set {0}".format(error))
+
+
+def get_all_iptable_rule_commands(command: str):
```

Comment:
there is something funny with this function: it is named "get_**all**_iptable_rule_commands", but it only gets ACCEPT rules?

Reply:
returns drop rule too.